### ISSUE TITLE
Bug with 404 returns on frontend routing due to the backend thinking they are endpoints

### ISSUE THREAD

**Issue by srimanachanta** [PhotonVision/photonvision#877]
The backend checks if any route is a backend endpoint before giving it to the frontend, this was found as a bug following #856 when HTML5 History was swapped to. This needs to be fixed by defining each Vue component as per Javalin's guide found [here](https://web.archive.org/web/20220124203740/https://javalin.io/tutorials/simple-frontends-with-javalin-and-vue).

**Comment by mdurrani808** [comment]
@srimanachanta good to close?

**Comment by srimanachanta** [comment]
> @srimanachanta good to close?

No, still need to resolve with the VueComponent API. Current fix is only temporary for this year's beta.

**Comment by Gold856** [comment]
We're not going to use the VueComponent API. We use more than just Vue components, including TypeScript files, and I don't really like web JARs that much. We're going to need a bundler either way, so we're instead going to use Javalin's SPA root config.
